poly-commitment/src/commitment.rs

```diff
@@ -424,65 +423,6 @@ pub fn combined_inner_product<G: CommitmentCurve>(
     res
 }
 
-enum OptShiftedPolynomial<P> {
-    Unshifted(P),
-    Shifted(P, usize),
-}
-
-/// A formal sum of the form
-/// `s_0 * p_0 + ... s_n * p_n`
-/// where each `s_i` is a scalar and each `p_i` is an optionally shifted polynomial.
-pub struct ChunkedPolynomial<F, P>(Vec<(F, OptShiftedPolynomial<P>)>);
-
-impl<F, P> Default for ChunkedPolynomial<F, P> {
-    fn default() -> ChunkedPolynomial<F, P> {
-        ChunkedPolynomial(vec![])
-    }
-}
-
-impl<F, P> ChunkedPolynomial<F, P> {
-    pub fn add_unshifted(&mut self, scale: F, p: P) {
-        self.0.push((scale, OptShiftedPolynomial::Unshifted(p)))
-    }
-
-    pub fn add_shifted(&mut self, scale: F, shift: usize, p: P) {
-        self.0
-            .push((scale, OptShiftedPolynomial::Shifted(p, shift)))
-    }
-}
-
-impl<'a, F: Field> ChunkedPolynomial<F, &'a [F]> {
-    pub fn to_dense_polynomial(&self) -> DensePolynomial<F> {
-        let mut res = DensePolynomial::<F>::zero();
-
-        let scaled: Vec<_> = self
-            .0
-            .par_iter()
-            .map(|(scale, segment)| {
-                let scale = *scale;
-                match segment {
-                    OptShiftedPolynomial::Unshifted(segment) => {
-                        let v = segment.par_iter().map(|x| scale * *x).collect();
-                        DensePolynomial::from_coefficients_vec(v)
-                    }
-                    OptShiftedPolynomial::Shifted(segment, shift) => {
-                        let mut v: Vec<_> = segment.par_iter().map(|x| scale * *x).collect();
-                        let mut res = vec![F::zero(); *shift];
-                        res.append(&mut v);
-                        DensePolynomial::from_coefficients_vec(res)
-                    }
-                }
-            })
-            .collect();
-
-        for p in scaled {
-            res += &p;
-        }
-
-        res
-    }
-}
-
```

Comment:
I actually think we should try not to touch this part of the code (poly-commitment) yet. As this logic seems a bit unnecessary to me at the moment. I think this PR could be more useful if targeted to the kimchi crate to begin with.

Reply:
> I actually think we should try not to touch this part of the code (poly-commitment) yet. As this logic seems a bit unnecessary to me at the moment. I think this PR could be more useful if targeted to the kimchi crate to begin with. Would you mind moving the poly-commitment stuff in this PR to another PR/branch for now?

Well, if we refactor `ChunkedPolynomial` into a new module, it is necessary that we move this part into the new file.